### PR TITLE
Completely remove TFM as a dependency and use reflections instead

### PR DESCRIPTION
**Pull request by Taahh**
Simplify removing color codes by using ChatColor#stripColor
Have an option to use sqlite or mysql, by default it will be sqlite
Remove TFM from the pom.xml and use reflections to access TFM methods to prevent a circular dependency

### CODE REVIEW

**Comment by Wild1145** [comment]
> ### [Codacy](https://app.codacy.com/gh/AtlasMediaGroup/TFGuilds/commit?cid=570324458) detected an issue:
> #### Message: `The utility class name 'Reflections' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'`
> #### Currently on:
> * **Commit**:  [11b3e19](https://github.com/AtlasMediaGroup/TFGuilds/commit/11b3e194ec59b834175eca7826a4813bc7c64ed4)
> * **File**:    [src/main/java/me/totalfreedom/tfguilds/util/Reflections.java](https://github.com/AtlasMediaGroup/TFGuilds/blob/11b3e194ec59b834175eca7826a4813bc7c64ed4/src/main/java/me/totalfreedom/tfguilds/util/Reflections.java)
> * **LineNum**: [6](https://github.com/AtlasMediaGroup/TFGuilds/blob/11b3e194ec59b834175eca7826a4813bc7c64ed4/src/main/java/me/totalfreedom/tfguilds/util/Reflections.java#L6)

Regarding this one, I think that's probably fine to ignore, it's a rule I think we turned off in the end on the main TFM Repo, but the first comment is one we need to resolve.

File: pom.xml [hunk 1]
```diff
@@ -4,9 +4,9 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>me.totalfreedom</groupId>
+    <groupId>com.github.AtlasMediaGroup</groupId>
     <artifactId>TFGuilds</artifactId>
-    <version>2021.06-RC3</version>
+    <version>master-4ec00d977c-1</version>
```
Comment (Wild1145) [inline pom.xml]: Could you change this back though, this should be our actual release version, not a commit ID from the master branch.

File: pom.xml [hunk 2]
```diff
@@ -72,11 +72,11 @@
             <version>1.16.5-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
-        <dependency>
+       <!-- <dependency>
```
Comment (Wild1145) [inline pom.xml]: If we're removing it can we remove rather than comment out please.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Taahh) [reply pom.xml]: changed

File: pom.xml (hunk 2 above)
Comment (Taahh) [reply pom.xml]: removed

File: src/main/java/me/totalfreedom/tfguilds/sql/SQLDatabase.java [hunk 3]
```diff
@@ -20,12 +22,23 @@ public SQLDatabase(TFGuilds plugin)
...
+                            ConfigEntry.MYSQL_HOST.getString(),
+                            ConfigEntry.MYSQL_PORT.getInteger(),
+                            ConfigEntry.MYSQL_DATABASE.getString()),
+                            ConfigEntry.MYSQL_USERNAME.getString(),
+                            password);
+                    break;
+                default:
+                    connection = DriverManager.getConnection("jdbc:sqlite:" + createDBFile(plugin).getAbsolutePath().replace("%20", " "));
+            }
+
```
Comment (Paldiu) [inline src/main/java/me/totalfreedom/tfguilds/sql/SQLDatabase.java:41]: This needs to be changed to prepared statements as this type of sql preparation can be vulnerable to sql injection.

**Review by Paldiu — CHANGES_REQUESTED** [review]

**Pull request merged**